File: katas/8 kyu/sum-mixed-array/solution.py
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
def sum_mix(arr):
  return sum(map(int , arr))
# ----------------------------------- TEST ----------------------------------- #
test.assert_equals(sum_mix([9, 3, '7', '3']), 22)
test.assert_equals(sum_mix(['5', '0', 9, 3, 2, 1, '9', 6, 7]), 42)
test.assert_equals(sum_mix(['3', 6, 6, 0, '5', 8, 5, '6', 2,'0']), 41)
test.assert_equals(sum_mix(['1', '5', '8', 8, 9, 9, 2, '3']), 45)
test.assert_equals(sum_mix([8, 0, 0, 8, 5, 7, 2, 3, 7, 8, 6, 7]), 61)

File: katas/8 kyu/sum-mixed-array/solution.php
<!-- sum-mixed-array -->
<?php
use PHPUnit\Framework\TestCase;
/* ------------------------------- MY SOLUTION ------------------------------ */
function sum_mix($a) {
  $b = array();
  foreach ($a as $item){
    array_push($b , intval($item));
  }
  return array_sum($b);
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
function sum_mix($a) {
  return array_sum($a);
}
/* ---------------------------------- TEST ---------------------------------- */
class SumMixTest extends TestCase {
  public function testExamples() {
    $this->assertEquals(22, sum_mix([9, 3, '7', '3']));
    $this->assertEquals(42, sum_mix(['5', '0', 9, 3, 2, 1, '9', 6, 7]));
    $this->assertEquals(41, sum_mix(['3', 6, 6, 0, '5', 8, 5, '6', 2, '0']));
  }
}